### PR TITLE
build workspace before trigger debug and support single file debug

### PR DESCRIPTION
**Pull request by ansyral**
Signed-off-by: xuzho <xuzho@microsoft.com>

### CODE REVIEW

**Comment by ansyral** [comment]
#94 
#32

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -79,10 +79,23 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 // log a warning message and continue, since logger failure should not block debug session
                 console.log("Cannot set log level to java debuggeer.")
             }
-            if (Object.keys(config).length === 0) { // No launch.json in current workspace.
-                // VSCode will create a launch.json automatically if no launch.json yet.
+
+            // trigger build workspace
+            try {
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Update the required vscode-java minimum version info in readme.

[the remark this answers is not in the document]
Comment (ansyral) [reply src/configurationProvider.ts]: nice catch! Could we specify the version in `extensionDependencies`  so vscode could upgrade language server to specified version if user upgrades debugger?

[the remark this answers is not in the document]
Comment (testforstephen) [reply src/configurationProvider.ts]: vscode doesn't support configuring the version for depended extension, can you check if the JAVA_BUILD_WORKSPACE command exists or not. If it doesn't exist, no need to execute the command.

[the remark this answers is not in the document]
Comment (ansyral) [reply src/configurationProvider.ts]: do we need to do that? already noted in readme that the requirement of ls is above 0.13.0

**Pull request merged**